arangod/MMFiles/MMFilesGeoIndex.cpp

```diff
@@ -330,7 +326,15 @@ Result MMFilesGeoIndex::insert(transaction::Methods& trx, LocalDocumentId const&
 
   IndexValue value(documentId, std::move(centroid));
   for (S2CellId cell : cells) {
+// The bool comperator is warned about in a unused code branch (which expects an int), MSVC doesn't properly detect this.
+#if (_MSC_VER >= 1)
+#pragma warning(push)
+#pragma warning( disable : 4804)
```

Comment:
Wouldn`t it be better not to duplicate code (as it will possibly lead to bugs related to editing just one branch of  #ifdef) ?

Reply:
yea, I was struggling here - more # noise, or clone that line...

Reply:
As I see, most times  #noise is preferred. To me it is more safe. So may be just stick to that?